sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/AbstractStatementParameterProcessor.java

```diff
@@ -0,0 +1,115 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.sql.parameterprocessor;
+
+import io.ballerina.runtime.api.types.Field;
+import io.ballerina.runtime.api.values.BObject;
+import org.ballerinalang.sql.exception.ApplicationError;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+/**
+ * This class has abstract implementation of methods required convert ballerina types into SQL types and
+ * other methods that process the parameters of the statement.
+ */
+abstract class AbstractStatementParameterProcessor {
+
+    protected abstract int getCustomOutParameterType(BObject typedValue) throws ApplicationError;
+    protected abstract int getCustomSQLType(BObject typedValue) throws ApplicationError;
+    protected abstract void setCustomSqlTypedParam(Connection connection, PreparedStatement preparedStatement,
+                            int index, BObject typedValue)
+            throws SQLException, ApplicationError, IOException;
+    protected abstract Object[] getCustomArrayData(Object value) throws ApplicationError;
+    protected abstract Object[] getCustomStructData(Object value, Connection conn) 
+            throws SQLException, ApplicationError;
+
+    protected abstract void setVarchar(int index, Object value, PreparedStatement preparedStatement)
+            throws SQLException;
+    protected abstract void setText(int index, Object value, PreparedStatement preparedStatement) throws SQLException;
```

Comment:
```suggestion
    protected abstract void setText(PreparedStatement preparedStatement, int index, Object value) throws SQLException;
```

Reply:
Shall we change the order of input parameters? 
1. PreparedStatement
2. sqlType
3. index
4. value

Reply:
I will change it.

Reply:
I added. Commit id is 116dd9c27696322fffcd1ca93d459e7db4b39ffb